plugins/ptp_operator/metrics/metrics.go

```diff
@@ -225,149 +308,288 @@ func (p *PTPEventManager) MockTest(t bool) {
 			if offsetFromMaster > threshold.MaxOffsetThreshold || offsetFromMaster < threshold.MinOffsetThreshold { // out of sync
-				p.PublishEvent(ceevent.FREERUN, offsetFromMaster, iface, channel.PTPEvent)
-				p.Stats[iface].lastClockState = ceevent.FREERUN
-				p.Stats[iface].lastOffset = int64(offsetFromMaster)
-				p.Stats[iface].reset()
+				p.PublishEvent(ceevent.FREERUN, offsetFromMaster, string(iface), channel.PTPEvent)
+				for _, i := range ifaces {
+					p.Stats[i].lastSyncState = ceevent.FREERUN
+					p.Stats[i].lastOffset = int64(offsetFromMaster)
+				}
 			} else {
-				p.Stats[iface].lastOffset = int64(offsetFromMaster)
-				p.Stats[iface].addValue(int64(offsetFromMaster)) // update off set when its in locked state and hold over only
+				for _, i := range ifaces {
+					p.Stats[i].lastOffset = int64(offsetFromMaster)
+					p.Stats[i].addValue(int64(offsetFromMaster)) // update off set when its in locked state and hold over only
+				}
 			}
 		case ceevent.HOLDOVER: // last state was holdover, since it is in locked now check if it qualifies to be locked
 			//do nothing , the time out will switch holdover to FREERUN
 		default: // not yet used states
-			log.Warnf("%s unkown handled last ptp state %s", iface, clockState)
-			p.PublishEvent(clockState, offsetFromMaster, iface, channel.PTPEvent) // change to locked
-			p.Stats[iface].lastClockState = clockState
-			p.Stats[iface].lastOffset = int64(offsetFromMaster)
+			log.Warnf("%s unkown handled last ptp state %s", iface, lastClockState)
+			p.PublishEvent(clockState, offsetFromMaster, string(iface), channel.PTPEvent) // change to locked
+			for _, i := range ifaces {
+				p.Stats[i].lastSyncState = clockState
+				p.Stats[i].lastOffset = int64(offsetFromMaster)
+			}
 			//do nothing already send event
 		}
 	case ceevent.FREERUN:
 		if offsetFromMaster < threshold.MaxOffsetThreshold && offsetFromMaster > threshold.MinOffsetThreshold { // within range
-			p.PublishEvent(ceevent.LOCKED, offsetFromMaster, iface, channel.PTPEvent) // change to locked
-			p.Stats[iface].lastClockState = ceevent.LOCKED
-			p.Stats[iface].lastOffset = int64(offsetFromMaster)
-			p.Stats[iface].addValue(int64(offsetFromMaster))
+			p.PublishEvent(ceevent.LOCKED, offsetFromMaster, string(iface), channel.PTPEvent) // change to locked
+			for _, i := range ifaces {
+				p.Stats[i].lastSyncState = ceevent.LOCKED
+				p.Stats[i].lastOffset = int64(offsetFromMaster)
+				p.Stats[i].addValue(int64(offsetFromMaster))
+			}
 		}
 	default:
 		log.Warnf("%s unkown current ptp state %s", iface, clockState)
-		p.PublishEvent(clockState, offsetFromMaster, iface, channel.PTPEvent) // change to locked
-		p.Stats[iface].lastClockState = clockState
-		p.Stats[iface].lastOffset = int64(offsetFromMaster)
-		p.Stats[iface].reset() // reset when its free run
+		p.PublishEvent(clockState, offsetFromMaster, string(iface), channel.PTPEvent) // change to locked
```

Comment:
comment says "change to locked" but the event published is unknown state (`clockState`).

Reply:
changed comment to unknown.
 this catches any state PTP produces that are unknown to us , since we are still exploring the ptp log events.